src/Panel/Panel.php

```diff
@@ -89,14 +543,160 @@ public static function link(App $kirby): bool
         return true;
     }
 
+    /**
+     * Returns the page object for the given id
+     *
+     * @param string $id Page's id
+     * @return \Kirby\Cms\Page|null
+     * @throws \Kirby\Exception\NotFoundException if the page cannot be found
+     */
+    public static function page(string $id)
+    {
+        $id   = str_replace('+', '/', $id);
+        $page = kirby()->page($id);
+
+        if ($page && $page->isReadable() === true) {
+            return $page;
+        }
+
+        throw new NotFoundException([
+            'key'  => 'page.notFound',
+            'data' => [
+                'slug' => $id
+            ]
+        ]);
+    }
+
+    /**
+     * Returns the model's object for the given path
+     *
+     * @param string $path Path to parent model
+     * @return \Kirby\Cms\Model|null
+     * @throws \Kirby\Exception\InvalidArgumentException if the model type is invalid
+     * @throws \Kirby\Exception\NotFoundException if the model cannot be found
+     */
+    public static function parent(string $path)
+    {
+        $modelType  = in_array($path, ['site', 'account']) ? $path : trim(dirname($path), '/');
+        $modelTypes = [
+            'site'    => 'site',
+            'users'   => 'user',
+            'pages'   => 'page',
+            'account' => 'account'
+        ];
+
+        $modelName = $modelTypes[$modelType] ?? null;
+
+        if (Str::endsWith($modelType, '/files') === true) {
+            $modelName = 'file';
+        }
```

Comment:
For easier understanding:

```suggestion
        $modelType = trim(dirname($path), '/');
        if (in_array($path, ['site', 'account']) === true) {
            $modelName = $path;
        } elseif (Str::endsWith($modelType, '/files') === true) {
            $modelName = 'file';
        } else {
            $modelTypes = [
                'users' => 'user',
                'pages' => 'page'
            ];

            $modelName = $modelTypes[$modelType] ?? null;
        }
```

Reply:
To be honest, I find yours much harder to read than Bastian's 🙈

Reply:
To be honest, I think both ways are not easy to read. Let's just keep it as it is.